docs/pages/testing/rspec.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module Pages
+	module Testing
+		class RSpec < ApplicationPage
+			def template
+				render Layout.new(title: "RSpec") do
+					render Markdown.new(<<~MD)
+						# RSpec
+
+						## Setup
+
+						Add test helpers to your suite:
+
+						```ruby
+						# spec/support/phlex.rb
+						require "phlex/test_helpers"
+
+						RSpec.configure do |config|
+							config.include Phlex::TestHelpers, type: :view
+						end
+						```
+
+						Remember to require this file in `rails_helper.rb` uncommenting the following line:
+
+						```ruby
+						Dir[Rails.root.join("spec/support/**/*.rb")].each { |f| require f }
+						```
+
+						If you are not using Rails, you just need to do this at your `spec/spec_helper.rb`:
+
+						```ruby
+						require "phlex/test_helpers"
+
+						RSpec.configure do |config|
+							# ... more
+
+							config.include Phlex::TestHelpers, type: :view
+						end
+						```
+
+						## Asserting without capybara
+
+						```ruby
+						# menu.rb
+						class Menu < Phlex::View
+							def template
+								ul do
+									li do
+										a(href: "/") { "Home" }
+										a(href: "/sign_in") { "Sign In" }
+									end
+								end
+							end
+						end
+
+						# menu_spec.rb
+						describe Menu, type: :view do
+							it "renders Home menu item" do
+								result = render_view(Menu.new)
+
+								selector = result.css('ul li a[href="/"]:contains("Home")')
+								expect(selector).to be_present
+							end
+						end
+						```
+
+						In addition to `css` you can use other Nokogiri finders like `xpath`, `at_css`, `at_xpath`. Read more about Nokogiri here: [https://nokogiri.org/](https://nokogiri.org/).
+						## Asserting with capybara
+						Setup capybara:
+
+						```ruby
+						# spec/support/capybara.rb
+						require "capybara/rspec"
+
+						RSpec.configure do |config|
+							config.include Capybara::RSpecMatchers, type: :view
+						end
+						```
+
+						More details here: [https://github.com/teamcapybara/capybara#using-capybara-with-rspec](https://github.com/teamcapybara/capybara#using-capybara-with-rspec)
+
+						You are now able to use capybara matchers:
+
+						```ruby
+						# menu_spec.rb
+						it "renders Home menu item" do
+							result = render_view(Menu.new)
+
+							expect(result).to have_link("Home", href: "/")
+						end
+						```
+
+						You can find more Capybara matchers here: [https://github.com/teamcapybara/capybara](https://github.com/teamcapybara/capybara).
+						### Passing blocks
+
+						You can pass blocks to `render_view`:
+
+						```ruby
+						it "renders items" do
+							result = render_view(Nav.new) do |nav|
+								nav.item(href: "/products/new") { "New Product" }
+								nav.item(href: "/categories/new") { "New Category" }
+							end
+
+							expect(result).to have_link("New Product", href: "/products/new")
+						end
```

Comment:
@stephannv Could you explain a bit more how this works? The `Nav` class is not defined anywhere on this PR and this looks like a cool usage, but I'm not sure what the `nav.item` methods are doing. Also, this documentation can no longer be found on the website.

Reply:
@davidalejandroaguilar this is just an example, but you could do this:

```ruby
class Nav < Phlex::HTML
  def template(&)
    ul class: "x y z" do
      yield_content(&)
    end
  end
  
  def item(href:, &)
    li class: "a b c" do
      a href: href do
        yield_content(&)
      end
    end
  end
end

render Nav.new do |nav|
  nav.item(href: "/example") { "Example" }
end
```

You can check this page (look all examples tabs too):
https://www.phlex.fun/html/components/